diff --git a/node_modules/@web3auth/no-modal/dist/noModal.cjs.js b/node_modules/@web3auth/no-modal/dist/noModal.cjs.js
index b8f0718..26e2844 100644
--- a/node_modules/@web3auth/no-modal/dist/noModal.cjs.js
+++ b/node_modules/@web3auth/no-modal/dist/noModal.cjs.js
@@ -165,14 +165,12 @@ class Web3AuthNoModal extends openlogin_jrpc_namespaceObject.SafeEventEmitter {
     this.cachedAdapter = null;
   }
   async addChain(chainConfig) {
-    if (this.status === base_namespaceObject.ADAPTER_STATUS.CONNECTED && this.connectedAdapterName) return this.walletAdapters[this.connectedAdapterName].addChain(chainConfig);
     if (this.commonJRPCProvider) {
       return this.commonJRPCProvider.addChain(chainConfig);
     }
     throw base_namespaceObject.WalletInitializationError.notReady(`No wallet is ready`);
   }
   async switchChain(params) {
-    if (this.status === base_namespaceObject.ADAPTER_STATUS.CONNECTED && this.connectedAdapterName) return this.walletAdapters[this.connectedAdapterName].switchChain(params);
     if (this.commonJRPCProvider) {
       return this.commonJRPCProvider.switchChain(params);
     }
diff --git a/node_modules/@web3auth/no-modal/dist/noModal.esm.js b/node_modules/@web3auth/no-modal/dist/noModal.esm.js
index 4e210b6..1c830fd 100644
--- a/node_modules/@web3auth/no-modal/dist/noModal.esm.js
+++ b/node_modules/@web3auth/no-modal/dist/noModal.esm.js
@@ -95,14 +95,12 @@ class Web3AuthNoModal extends SafeEventEmitter {
     this.cachedAdapter = null;
   }
   async addChain(chainConfig) {
-    if (this.status === ADAPTER_STATUS.CONNECTED && this.connectedAdapterName) return this.walletAdapters[this.connectedAdapterName].addChain(chainConfig);
     if (this.commonJRPCProvider) {
       return this.commonJRPCProvider.addChain(chainConfig);
     }
     throw WalletInitializationError.notReady(`No wallet is ready`);
   }
   async switchChain(params) {
-    if (this.status === ADAPTER_STATUS.CONNECTED && this.connectedAdapterName) return this.walletAdapters[this.connectedAdapterName].switchChain(params);
     if (this.commonJRPCProvider) {
       return this.commonJRPCProvider.switchChain(params);
     }
diff --git a/node_modules/@web3auth/no-modal/src/noModal.ts b/node_modules/@web3auth/no-modal/src/noModal.ts
index 2a35915..fbe7e33 100644
--- a/node_modules/@web3auth/no-modal/src/noModal.ts
+++ b/node_modules/@web3auth/no-modal/src/noModal.ts
@@ -189,9 +189,6 @@ export class Web3AuthNoModal extends SafeEventEmitter implements IWeb3Auth {
   }
 
   public async addChain(chainConfig: CustomChainConfig): Promise<void> {
-    if (this.status === ADAPTER_STATUS.CONNECTED && this.connectedAdapterName)
-      return this.walletAdapters[this.connectedAdapterName].addChain(chainConfig);
-
     if (this.commonJRPCProvider) {
       return this.commonJRPCProvider.addChain(chainConfig);
     }
@@ -199,9 +196,6 @@ export class Web3AuthNoModal extends SafeEventEmitter implements IWeb3Auth {
   }
 
   public async switchChain(params: { chainId: string }): Promise<void> {
-    if (this.status === ADAPTER_STATUS.CONNECTED && this.connectedAdapterName)
-      return this.walletAdapters[this.connectedAdapterName].switchChain(params);
-
     if (this.commonJRPCProvider) {
       return this.commonJRPCProvider.switchChain(params);
     }
